### PR TITLE
#1044 ☑️ Percent Tumour Validation

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

Validates Specimen files where 'percent_tumour_cells_measurement_method' = 'Not Applicable' contain no values in the 'percent_tumour_cells' field

**Type of Change**

- [ ] Bug
- [ ] Refactor
- [ ] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/submission/validation-clinical/specimen.ts [hunk 1]
```diff
@@ -81,9 +81,36 @@ export const validate = async (
     checkTimeConflictWithDonor(donorDataToValidateWith, specimenRecord, errors);
   }
 
+  validatePercentTumour(specimenRecord, errors);
+
   return { errors };
 };
 
+const validatePercentTumour = (
```
Comment (justincorrigible) [inline src/submission/validation-clinical/specimen.ts:89]: I see the pattern has already been established here before this PR, but just to document the though, in JavaScript it is considered unidiomatic to use/execute a function before it's been define and implemented.

i.e. this line 89 should be happening some line before 84, and the same goes for `checkTimeConflictWithDonor`, etc.

File: test/unit/submission/validation.spec.ts [hunk 2]
```diff
@@ -1500,6 +1500,155 @@ describe('data-validator', () => {
       chai.expect(result[ClinicalEntitySchemaNames.SPECIMEN].dataErrors.length).to.eq(0);
     });
 
+    it('should validate Specimen Percent Tumour is blank when Tumour Measurement Method is `Not Applicable`', async () => {
+      const existingDonorAB2Mock: Donor = stubs.validation.existingDonor11();
+      const submittedAB1Records = {};
```
Comment (justincorrigible) [inline test/unit/submission/validation.spec.ts:1505]:
```suggestion
      const submittedAB1Records = {};

```

File: test/unit/submission/validation.spec.ts [hunk 3]
```diff
@@ -1500,6 +1500,155 @@ describe('data-validator', () => {
...
+        .to.deep.include(specimenMeasurementErr);
+
+      chai
+        .expect(result[ClinicalEntitySchemaNames.SPECIMEN].dataErrors)
+        .to.deep.include(specimenMeasurementErr2);
+    });
+
+    it('should allow blank Specimen Percent Tumour Cells field when Tumour Measurement Method is `Not Applicable`', async () => {
+      const existingDonorAB1Mock: Donor = stubs.validation.existingDonor11();
+      const submittedAB1Records = {};
```
Comment (justincorrigible) [inline test/unit/submission/validation.spec.ts:1605]:
```suggestion
      const submittedAB1Records = {};

```

File: test/unit/submission/validation.spec.ts [hunk 4]
```diff
@@ -1500,6 +1500,155 @@ describe('data-validator', () => {
...
+      ClinicalSubmissionRecordsOperations.addRecord(
+        ClinicalEntitySchemaNames.DONOR,
+        submittedAB1Records,
+        {
+          [SampleRegistrationFieldsEnum.submitter_donor_id]: 'AB2',
+          [SampleRegistrationFieldsEnum.program_id]: 'PEME-CA',
+          [DonorFieldsEnum.vital_status]: 'Alive',
+          index: 0,
+        },
+      );
```
Comment (justincorrigible) [inline test/unit/submission/validation.spec.ts:1615]: let's keep spacing consistent
```suggestion
      );

```

**Review by justincorrigible — APPROVED** [review]

File: src/submission/validation-clinical/specimen.ts (hunk 1 above)
Comment (demariadaniel) [reply src/submission/validation-clinical/specimen.ts:41]: Updated. All of the validation files are written this way. Makes me wonder how old the files are.

File: test/unit/submission/validation.spec.ts (hunk 4 above)
Comment (justincorrigible) [reply test/unit/submission/validation.spec.ts:1617]: much appreciated!

**Review by justincorrigible — APPROVED** [review]

**Pull request merged**